### PR TITLE
Introduce SurfaceStep

### PR DESCRIPTION
**Pull request by brownd1978**
SurfaceStep is a new MC truth class that functions like StrawGasStep, CaloShowerStep, etc, but for passive materials and virtual detectors. It uses the SurfaceId class to define the surfaces, and summarizes the information contained in contiguous StepPointMC objects. This greatly reduced the payload and renders the output information more useful for physics studies, and insulates the results from the details of the G4 stepping parameters. This PR introduces the class itself plus producer, diag module, printer, and default production configuration.

Note that SurfaceId was moved to DataProducts (from RecoDataProducts and KinKalGeom) to allow this to work. It's fine, as the surface concept isn't specific to reco or mc.

### CODE REVIEW

**Review by rlcee — APPROVED** [review]
- "surface" would be more clear as "material" or "volume"
- split SurfaceId from enum
- SurfaceId::index could be named more clearly and commented
- fix include guards in SurfaceId
- print looks fine

**Comment by brownd1978** [comment]
> * "surface" would be more clear as "material" or "volume"
Surface is explicit in the downstream use cases, particularly reconstruction. 'volume' or 'material' are less accurate.
> * split SurfaceId from enum
These classes don't make sense as independent objects, just like the detail of the enum class it is clearer to keep them in 1 file
> * SurfaceId::index could be named more clearly and commented
I added some comments. The name must be general.
> * fix include guards in SurfaceId
done
> * print looks fine

Note too SurfaceId pre-exists this PR by over a year, this PR simply moves it.

**Review by kutschke — COMMENTED** [review]
I saw comment a comment in the conversation (or an issue?) that explains that index class distinguishes items with multiple placements, such as the target foils.  I don't see it explained in any of the code files.  Did I miss it?  If not, please add a comment to SurfaceId.hh .  Also, are there other entities that have multiple indices?

File: CommonMC/fcl/prolog.fcl [hunk 1]
```diff
@@ -26,6 +26,21 @@ CommonMC: {
       RecoFromMCTruthErr : 0.5
     }
   }
+
```
[the remark this answers is not in the document]
Comment (brownd1978) [reply CommonMC/fcl/prolog.fcl:29]: This is historical baggage that needs to be cleaned up in the Production repo as well.  I will do it in a separate PR.

[the remark this answers is not in the document]
Comment (kutschke) [reply CommonMC/fcl/prolog.fcl:20]: OK.  Thanks

**Review by kutschke — APPROVED** [review]
Thanks for making the changes.

**Comment by kutschke** [comment]
@sdifalco a gentle reminder that this PR is on your to-review list.

**Comment by kutschke** [comment]
@FNALbuild run build test

**Comment by kutschke** [comment]
Needs to test with a Production PR

@FNALbuild run build test with Mu2e/Production#329

**Comment by brownd1978** [comment]
AFAIK I've addressed all comments and have no further plans to work on this. Stefano was added as reviewer just to keep him in the loop.

**Comment by sdifalco** [comment]
> @sdifalco a gentle reminder that this PR is on your to-review list.

Sorry but I am been (and still am) very busy with funding requests to INFN in these days. Thanks for taking me in the loop. I will approve what you have already commented. Thanks Ed for the nice job!

**Comment by kutschke** [comment]
> AFAIK I've addressed all comments and have no further plans to work on this. Stefano was added as reviewer just to keep him in the loop.

In that case it's ready to merge.  Do you expect this to have impact on the nightly validation?  The STM PR that I merged earlier does modify the G4 geometry but only in the STM region.

**Review by sdifalco — APPROVED** [review]

**Comment by kutschke** [comment]
I will go ahead and merge now under the assumption that if there are issues to sort out with the nightly, we can do it tomoorow

**Pull request merged**